clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1088,14 +1088,24 @@ public void run() {
                             markCoordinatorUnknown();
                         } else if (heartbeat.pollTimeoutExpired(now)) {
                             // the poll timeout has expired, which means that the foreground thread has stalled
-                            // in between calls to poll(), so we explicitly leave the group.
-                            log.warn("This member will leave the group because consumer poll timeout has expired. This " +
-                                    "means the time between subsequent calls to poll() was longer than the configured " +
-                                    "max.poll.interval.ms, which typically implies that the poll loop is spending too " +
-                                    "much time processing messages. You can address this either by increasing " +
-                                    "max.poll.interval.ms or by reducing the maximum size of batches returned in poll() " +
-                                    "with max.poll.records.");
-                            maybeLeaveGroup();
+                            // in between calls to poll().
+                            if (isDynamicMember()) {
+                                log.warn("This dynamic member will leave the group because consumer poll timeout has expired. This " +
```

Comment:
I think the motivation of this warning entry is to educate users to increase the corresponding configs, and hence not mentioning that it will leave group is fine. How about we just re-word it a bit to sth like

```
This member is at the risk of leave the group because ...
```

And then we do not need to branch here anymore?

Reply:
sg!

Reply:
Actually nvm.. I realized the true issue is that we want to avoid WARN log spamming for static members, and my previous comment did not resolve it at all.. Here's another idea: we can pass in a string parameter `rootCause` into `maybeLeaveGroup`, and then in the info logging we can augment it to 

```
Member {} sending LeaveGroup request to coordinator {} due to {}
```

And then in this place we can pass in the whole statement educating users to re-configure; and in other cases the root-cause can just be `due to being closed` and `due to unsubscribing`.

Reply:
Sounds good, updated.